server/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -394,15 +409,17 @@ protected static void innerToXContent(
             builder.field(STACK_TRACE, ExceptionsHelper.stackTrace(throwable));
         }
 
-        Throwable[] allSuppressed = throwable.getSuppressed();
-        if (allSuppressed.length > 0) {
-            builder.startArray(SUPPRESSED.getPreferredName());
-            for (Throwable suppressed : allSuppressed) {
-                builder.startObject();
-                generateThrowableXContent(builder, params, suppressed);
-                builder.endObject();
+        if (nestedLevel <= MAX_NESTED_EXCEPTION_LEVEL) {
```

Comment:
We don't limit the suppressed exceptions themselves in the stream serialization, we leave it up to the recursive serialization call. Is this needed for anything specific, or can we leave the check to the `generateThrowableXContent` call below?

Reply:
It was needed because I was trying to put the depth check in the wrong place - I was doing it at the point of printing the cause of the exception that hit the limit, not the place where we try to print the exception itself.
It all became cleaner once I realised I was making that mistake